File: take_screenshots.py
#!/usr/bin/env python3
"""
Screenshot script for Wander web app using Python playwright.
"""
import asyncio
import os
import sys
from pathlib import Path

from playwright.async_api import async_playwright

ASSETS_DIR = r"C:\Users\vjbel\hacks\wander\assets"
BASE_URL = "http://localhost:3000"

async def run():
    os.makedirs(ASSETS_DIR, exist_ok=True)
    
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(
            viewport={"width": 1440, "height": 900}
        )
        page = await context.new_page()
        
        # STEP 1: Open landing page
        print("STEP 1: Opening http://localhost:3000...")
        await page.goto(BASE_URL, wait_until="networkidle", timeout=30000)
        await page.wait_for_timeout(2000)
        await page.screenshot(path=os.path.join(ASSETS_DIR, "01-landing.png"), full_page=True)
        print(f"Saved: {os.path.join(ASSETS_DIR, '01-landing.png')}")
        
        # Check what the page looks like - get title and any visible text
        title = await page.title()
        print(f"Page title: {title}")
        
        # STEP 2: Fill in the form
        print("\nSTEP 2: Filling in the form...")
        
        # Try to find the "Starting from" field
        start_selectors = [
            'input[placeholder*="Starting"]',
            'input[placeholder*="starting"]', 
            'input[placeholder*="Start"]',
            'input[placeholder*="from"]',
            'input[placeholder*="From"]',
            'input[name*="start"]',
            'input[name*="from"]',
            'input[id*="start"]',
            'input[id*="from"]',
        ]
        
        start_field = None
        for sel in start_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    start_field = el
                    print(f"Found start field with: {sel}")
                    break
            except:
                pass
        
        if start_field:
            await start_field.click()
            await start_field.fill("Hell's Kitchen, NYC")
            await page.wait_for_timeout(500)
        else:
            print("WARNING: Could not find start field, trying all inputs...")
            inputs = page.locator('input[type="text"], input:not([type])').all()
            inp_list = await inputs
            if len(inp_list) > 0:
                await inp_list[0].click()
                await inp_list[0].fill("Hell's Kitchen, NYC")
                print("Filled first input")
        
        # Try to find the "Ending up at" field
        end_selectors = [
            'input[placeholder*="Ending"]',
            'input[placeholder*="ending"]',
            'input[placeholder*="End"]', 
            'input[placeholder*="Destination"]',
            'input[placeholder*="destination"]',
            'input[placeholder*="to"]',
            'input[placeholder*="To"]',
            'input[name*="end"]',
            'input[name*="to"]',
            'input[id*="end"]',
            'input[id*="to"]',
        ]
        
        end_field = None
        for sel in end_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    end_field = el
                    print(f"Found end field with: {sel}")
                    break
            except:
                pass
        
        if end_field:
            await end_field.click()
            await end_field.fill("Flatiron District, NYC")
            await page.wait_for_timeout(500)
        else:
            print("WARNING: Could not find end field, trying second input...")
            inputs_list = await page.locator('input[type="text"], input:not([type])').all()
            if len(inputs_list) > 1:
                await inputs_list[1].click()
                await inputs_list[1].fill("Flatiron District, NYC")
                print("Filled second input")
        
        # Click the 'Green & Scenic' vibe button
        vibe_selectors = [
            'button:has-text("Green & Scenic")',
            'button:has-text("Green")',
            'button:has-text("Scenic")',
            '[class*="vibe"]:has-text("Green")',
            '*:has-text("🌿")',
            'button:has-text("🌿")',
        ]
        
        vibe_clicked = False
        for sel in vibe_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    await el.click(timeout=5000)
                    vibe_clicked = True
                    print(f"Clicked vibe button with: {sel}")
                    break
            except Exception as e:
                print(f"  Failed {sel}: {e}")
        
        if not vibe_clicked:
            print("WARNING: Could not click vibe button")
        
        await page.wait_for_timeout(800)
        await page.screenshot(path=os.path.join(ASSETS_DIR, "02-form-filled.png"), full_page=True)
        print(f"Saved: {os.path.join(ASSETS_DIR, '02-form-filled.png')}")
        
        # STEP 3: Click the submit button
        print("\nSTEP 3: Clicking Begin wandering button...")
        submit_selectors = [
            'button:has-text("Begin wandering")',
            'button:has-text("wandering")',
            'button:has-text("Wander")',
            'button:has-text("Start")',
            'button[type="submit"]',
            'input[type="submit"]',
        ]
        
        submit_clicked = False
        for sel in submit_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    await el.click(timeout=5000)
                    submit_clicked = True
                    print(f"Clicked submit with: {sel}")
                    break
            except Exception as e:
                print(f"  Failed {sel}: {e}")
        
        if not submit_clicked:
            print("WARNING: Could not find submit button, pressing Enter...")
            await page.keyboard.press("Enter")
        
        await page.wait_for_timeout(2000)
        await page.screenshot(path=os.path.join(ASSETS_DIR, "03-loading.png"), full_page=True)
        print(f"Saved: {os.path.join(ASSETS_DIR, '03-loading.png')}")
        
        # STEP 4: Wait for route to load (up to 30 seconds)
        print("\nSTEP 4: Waiting for route to load...")
        try:
            # Wait for loading to finish - look for content that indicates route is loaded
            # Try waiting for the loading indicator to disappear
            await page.wait_for_timeout(5000)
            
            # Try to wait for route-specific content
            route_loaded = False
            for _ in range(6):  # Try for 30 seconds (6 x 5s)
                content = await page.content()
                # Check for common route indicators
                if any(term in content.lower() for term in ['stop', 'route', 'walk', 'visit']):
                    # Check if loading indicator is gone
                    loading_visible = False
                    try:
                        loading_el = page.locator('text="Reading the streets", text="Loading", [class*="loading"], [class*="spinner"]').first
                        loading_visible = await loading_el.is_visible(timeout=1000)
                    except:
                        loading_visible = False
                    
                    if not loading_visible:
                        route_loaded = True
                        print("Route appears loaded!")
                        break
                
                print("Still loading, waiting 5 more seconds...")
                await page.wait_for_timeout(5000)
        except Exception as e:
            print(f"Error waiting for route: {e}")
        
        await page.wait_for_timeout(2000)
        await page.screenshot(path=os.path.join(ASSETS_DIR, "04-route.png"), full_page=True)
        print(f"Saved: {os.path.join(ASSETS_DIR, '04-route.png')}")
        
        # Get route title
        route_title = "Unknown"
        title_selectors = ['h1', 'h2', '[class*="route-title"]', '[class*="title"]', '[class*="heading"]']
        for sel in title_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    route_title = await el.inner_text()
                    if route_title.strip():
                        print(f"Route title: {route_title}")
                        break
            except:
                pass
        
        # STEP 5: Click Insider tip button
        print("\nSTEP 5: Looking for Insider tip button...")
        tip_selectors = [
            'button:has-text("Insider tip")',
            'button:has-text("insider tip")',
            'button:has-text("Insider Tip")',
            'button:has-text("tip")',
            '[class*="tip"] button',
            'button[class*="tip"]',
            '*:has-text("Insider")',
        ]
        
        tip_clicked = False
        for sel in tip_selectors:
            try:
                el = page.locator(sel).first
                count = await el.count()
                if count > 0:
                    await el.scroll_into_view_if_needed()
                    await page.wait_for_timeout(500)
                    await el.click(timeout=8000)
                    tip_clicked = True
                    print(f"Clicked tip button with: {sel}")
                    break
            except Exception as e:
                print(f"  Failed {sel}: {e}")
        
        if not tip_clicked:
            print("WARNING: Could not find Insider tip button")
        
        await page.wait_for_timeout(1000)
        await page.screenshot(path=os.path.join(ASSETS_DIR, "05-insider-tip.png"), full_page=True)
        print(f"Saved: {os.path.join(ASSETS_DIR, '05-insider-tip.png')}")
        
        await browser.close()
        
        print("\n=== FINAL SUMMARY ===")
        print(f"Route title: {route_title}")
        print(f"All screenshots saved to: {ASSETS_DIR}")
        print("Files saved:")
        for fname in ["01-landing.png", "02-form-filled.png", "03-loading.png", "04-route.png", "05-insider-tip.png"]:
            fpath = os.path.join(ASSETS_DIR, fname)
            if os.path.exists(fpath):
                size = os.path.getsize(fpath)
                print(f"  ✓ {fpath} ({size:,} bytes)")
            else:
                print(f"  ✗ {fpath} (NOT FOUND)")

if __name__ == "__main__":
    asyncio.run(run())

File: take_screenshots.js
const { chromium } = require('playwright');
const path = require('path');

const ASSETS_DIR = 'C:\\Users\\vjbel\\hacks\\wander\\assets';

async function run() {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 1440, height: 900 },
  });
  const page = await context.newPage();

  // STEP 1: Open landing page
  console.log('STEP 1: Opening http://localhost:3000...');
  await page.goto('http://localhost:3000', { waitUntil: 'networkidle', timeout: 30000 });
  await page.waitForTimeout(1500);
  await page.screenshot({ path: path.join(ASSETS_DIR, '01-landing.png'), fullPage: true });
  console.log('Saved: 01-landing.png');

  // STEP 2: Fill in the form
  console.log('STEP 2: Filling in the form...');

  // Fill "Starting from" field
  const startField = page.locator('input[placeholder*="Starting"], input[placeholder*="starting"], input[placeholder*="Start"], input[placeholder*="From"], input[placeholder*="from"]').first();
  await startField.click();
  await startField.fill("Hell's Kitchen, NYC");
  await page.waitForTimeout(500);

  // Fill "Ending up at" field
  const endField = page.locator('input[placeholder*="Ending"], input[placeholder*="ending"], input[placeholder*="End"], input[placeholder*="To"], input[placeholder*="to"], input[placeholder*="Destination"]').first();
  await endField.click();
  await endField.fill('Flatiron District, NYC');
  await page.waitForTimeout(500);

  // Click the 'Green & Scenic' vibe button
  const vibeButton = page.locator('button:has-text("Green"), button:has-text("Scenic"), [role="button"]:has-text("Green")').first();
  try {
    await vibeButton.click({ timeout: 5000 });
    console.log('Clicked Green & Scenic vibe button');
  } catch (e) {
    console.log('Could not find Green & Scenic button by text, trying emoji approach...');
    // Try finding by emoji
    const emojiButton = page.locator('button:has-text("🌿")').first();
    try {
      await emojiButton.click({ timeout: 5000 });
      console.log('Clicked 🌿 button');
    } catch (e2) {
      console.log('Could not click vibe button:', e2.message);
    }
  }
  await page.waitForTimeout(500);

  await page.screenshot({ path: path.join(ASSETS_DIR, '02-form-filled.png'), fullPage: true });
  console.log('Saved: 02-form-filled.png');

  // STEP 3: Click "Begin wandering" button
  console.log('STEP 3: Clicking Begin wandering button...');
  const submitBtn = page.locator('button:has-text("Begin wandering"), button:has-text("Wander"), button:has-text("wandering"), button[type="submit"]').first();
  await submitBtn.click();
  await page.waitForTimeout(1500);
  await page.screenshot({ path: path.join(ASSETS_DIR, '03-loading.png'), fullPage: true });
  console.log('Saved: 03-loading.png');

  // STEP 4: Wait for route to load (up to 30 seconds)
  console.log('STEP 4: Waiting for route to load...');
  try {
    // Wait for route content - look for numbered stops or route title
    await page.waitForSelector(
      'h1, h2, .route-title, [class*="route"], [class*="stop"], [class*="card"]',
      { timeout: 30000 }
    );
    // Additional wait to let everything render
    await page.waitForTimeout(2000);
  } catch (e) {
    console.log('Timeout waiting for route, taking screenshot anyway...');
    await page.waitForTimeout(3000);
  }
  await page.screenshot({ path: path.join(ASSETS_DIR, '04-route.png'), fullPage: true });
  console.log('Saved: 04-route.png');

  // Get the route title
  let routeTitle = 'Unknown';
  try {
    const titleEl = await page.locator('h1, h2, [class*="title"], [class*="route-name"]').first();
    routeTitle = await titleEl.innerText();
    console.log('Route title:', routeTitle);
  } catch (e) {
    console.log('Could not get route title');
  }

  // STEP 5: Click an "Insider tip" button
  console.log('STEP 5: Looking for Insider tip button...');
  try {
    const insiderBtn = page.locator('button:has-text("Insider tip"), button:has-text("insider"), button:has-text("tip"), [class*="tip"]').first();
    await insiderBtn.scrollIntoViewIfNeeded();
    await insiderBtn.click({ timeout: 10000 });
    await page.waitForTimeout(1000);
    await page.screenshot({ path: path.join(ASSETS_DIR, '05-insider-tip.png'), fullPage: true });
    console.log('Saved: 05-insider-tip.png');
  } catch (e) {
    console.log('Could not find/click Insider tip button:', e.message);
    // Take screenshot anyway
    await page.screenshot({ path: path.join(ASSETS_DIR, '05-insider-tip.png'), fullPage: true });
    console.log('Saved: 05-insider-tip.png (no tip expanded)');
  }

  await browser.close();

  console.log('\n=== SUMMARY ===');
  console.log('Route title found:', routeTitle);
  console.log('All screenshots saved to:', ASSETS_DIR);
}

run().catch(err => {
  console.error('ERROR:', err);
  process.exit(1);
});
